src/main/java/edu/hm/hafner/coverage/parser/CloverParser.java

```diff
@@ -0,0 +1,234 @@
+import edu.hm.hafner.coverage.Metric;
+import edu.hm.hafner.coverage.ModuleNode;
+import edu.hm.hafner.coverage.Node;
+import edu.hm.hafner.coverage.PackageNode;
+import edu.hm.hafner.util.FilteredLog;
+import edu.hm.hafner.util.SecureXmlParserFactory;
+import edu.hm.hafner.util.SecureXmlParserFactory.ParsingException;
+import edu.hm.hafner.util.TreeString;
+
+/**
+ * Parses Clover reports into a hierarchical Java Object Model.
+ */
+public class CloverParser extends CoverageParser {
+    private static final long serialVersionUID = -1903059983931698657L;
+
+    private static final QName COVERAGE = new QName("coverage");
+    private static final QName PROJECT = new QName("project");
+    private static final QName PACKAGE = new QName("package");
+    private static final QName METRICS = new QName("metrics");
+    private static final QName FILE = new QName("file");
+    private static final QName NAME = new QName("name");
+    private static final QName PATH = new QName("path");
+    private static final QName STATEMENTS = new QName("statements");
+    private static final QName COVERED_STATEMENTS = new QName("coveredstatements");
+    private static final QName CONDITIONALS = new QName("conditionals");
+    private static final QName COVERED_CONDITIONALS = new QName("coveredconditionals");
+    private static final QName LINE = new QName("line");
+    private static final QName NUM = new QName("num");
+    private static final QName COUNT = new QName("count");
+
+    /**
+     * Creates a new instance of {@link CloverParser}.
+     *
+     * @param processingMode
+     *         determines whether to ignore errors
+     */
+    public CloverParser(final ProcessingMode processingMode) {
+        super(processingMode);
+    }
+
+    @Override
+    protected ModuleNode parseReport(final Reader reader, final String fileName, final FilteredLog log) {
+        try {
+            var factory = new SecureXmlParserFactory();
+            var eventReader = factory.createXmlEventReader(reader);
+
+            while (eventReader.hasNext()) {
+                XMLEvent event = eventReader.nextEvent();
+                if (event.isStartElement()) {
+                    var startElement = event.asStartElement();
+                    var tagName = startElement.getName();
+                    if (COVERAGE.equals(tagName)) {
+                        ModuleNode root = new ModuleNode("");
+                        if (!readCoverage(eventReader, root, fileName).hasChildren()) {
+                            handleEmptyResults(fileName, log);
+                        }
+                        else {
+                            return root;
+                        }
+                    }
+                }
+            }
+            handleEmptyResults(fileName, log);
+            return new ModuleNode("empty");
+        }
+        catch (XMLStreamException exception) {
+            throw new SecureXmlParserFactory.ParsingException(exception);
```

Comment:
```suggestion
            throw new ParsingException(exception);
```
Then I can simpler move the exception in a followup PR.